share/python/g3mp2.py

```diff
@@ -0,0 +1,685 @@
+    ZPE(HF/6-31G(d))=     0.020516   ZPE Scale Factor =     0.892900
+    HLC             =    -0.036680   Free Energy      =     0.004735
+    Thermal Energy  =     0.025811   Thermal Enthalpy =     0.026755
+    ________________________________________________________________
+    E(G3(MP2)) @ 0K =   -76.341911   E(G3(MP2)) @298K =   -76.339077
+    H(G3(MP2))      =   -76.338133   G(G3(MP2))       =   -76.360152
+    ________________________________________________________________
+      HEAT OF FORMATION  (0K):     -56.87 kCal/mol
+      HEAT OF FORMATION(298K):     -57.42 kCal/mol
+    ________________________________________________________________
+    '''
+
+    if use_QCISDT is True:
+        ccstring = 'QCISDT '
+    else:
+        ccstring = 'CCSD(T)'
+
+    report('    ____________________________________________________________Psi4')
+    report('               SUMMARY OF G3(MP2, %s) CALCULATIONS'
+           % ccstring)
+    report('    ________________________________________________________________')
+    report('    MP2/6-31G(d)    = %12.6f   %s/6-31G(d) = %12.6f'
+           % (Emp2_fc_631gd, ccstring, Ecc))
+    report('    MP2/G3MP2large  = %12.6f   delta(MP2)       = %12.6f'
+           % (Eg3mp2large, dMP2))
+    report('    ZPE(HF/6-31G(d))= %12.6f   ZPE Scale Factor = %12.6f'
+           % (Ezpe_scaled, 0.8929))
+    report('    HLC             = %12.6f   Free Energy      = %12.6f'
+           % (Ehlc, Gthermal))
+    report('    Thermal Energy  = %12.6f   Thermal Enthalpy = %12.6f'
+           % (Ethermal, Hthermal))
+    report('    ________________________________________________________________')
+    report('    E(G3(MP2)) @ 0K = %12.6f   E(G3(MP2)) @298K = %12.6f'
+           % (E0_g3mp2, E298_g3mp2))
+    report('    H(G3(MP2))      = %12.6f   G(G3(MP2))       = %12.6f'
+           % (H298_g3mp2, G298_g3mp2))
+    report('    ________________________________________________________________')
+    report("          HEAT OF FORMATION  (0K): % 10.2f kCal/mol" % dHf0)
+    report("          HEAT OF FORMATION(298K): % 10.2f kCal/mol" % dHf298)
+    report('    ________________________________________________________________')
+
+    # write final energies in CSV format for trivial extraction.
+    # A database of several molecules may be roughly constructed by common
+    # shell tools, e.g.,
+    #
+    #   grep -A 3 "CSV format" *.out |grep -v "CSV format" >> tempcsv.txt
+    #   sort -r <tempcsv.txt | uniq >moldb.csv
+
+    log("\n")
+    log("## CSV format of G3(MP2) energies")
+
+    log("#specie, E0, E298, H298, Gibbs, dHf0, dHf298")
+    log("#unit, Ha, Ha, Ha, Ha, kcal, kcal")
+    log("\"%s\",%.6f,%.6f,%.6f,%.6f,%.2f,%.2f" %
+        (g3mp2_name,
+         E0_g3mp2, E298_g3mp2,
+         H298_g3mp2, G298_g3mp2,
+         dHf0, dHf298))
+
+    log("\n")
+
+    psi4.clean()
+    optstash.restore()
+
+    # return E @0K g3mp2 results
+
+    return E0_g3mp2
```

Comment:
Any method that returns from `procedures` must return a Wavefunction. In same cases (like this one) its not entirely clear what that wavefunction should be. Please select the most appropriate Wavefunction to return.

Also, set `psi4.set_variable('CURRENT ENERGY', E0_g3mp2)` so this particular energy is available globally.